### PR TITLE
[PM-11490] iOS 18 fix broken account switcher

### PR DESCRIPTION
**Pull request by fedemkr**
## 🎟️ Tracking

[PM-11490](https://bitwarden.atlassian.net/browse/PM-11490)

## 📔 Objective

iOS 18 -> Fix broken account switcher that was being shown behind the toolbar.

Removed `scrollOffsetChanged` and `scrollOffset` from the `ProfileSwitcherHandler`/`ProfileSwitcherState`/`ProfileSwitcherAction` as they're no longer necessary as the state is now handled by the `ProfileSwitcherView` -> Thanks @matt-livefront for this fix/improvement.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-11490]: https://bitwarden.atlassian.net/browse/PM-11490?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: BitwardenShared/UI/Auth/ProfileSwitcher/ProfileSwitcherView.swift [hunk 1]
```diff
@@ -67,7 +61,7 @@ struct ProfileSwitcherView: View {
     private var backgroundView: some View {
         VStack {
             Asset.Colors.backgroundPrimary.swiftUIColor
-                .frame(height: abs(min(store.state.scrollOffset.y, 0)))
+                .frame(height: store.state.isVisible ? 0.01 : 0)
```
Comment (phil-livefront) [inline BitwardenShared/UI/Auth/ProfileSwitcher/ProfileSwitcherView.swift:64]: was this needed?

**Review by phil-livefront — DISMISSED** [review]

File: BitwardenShared/UI/Auth/ProfileSwitcher/ProfileSwitcherView.swift [hunk 2]
```diff
@@ -9,13 +9,7 @@ struct ProfileSwitcherView: View {
...
 
     var body: some View {
-        OffsetObservingScrollView(
-            axes: store.state.isVisible ? .vertical : [],
-            offset: .init(
-                get: { store.state.scrollOffset },
-                set: { store.send(.scrollOffsetChanged($0)) }
-            )
-        ) {
+        ScrollView {
```
Comment (matt-livefront) [inline BitwardenShared/UI/Auth/ProfileSwitcher/ProfileSwitcherView.swift:12]: ❓ I thought @phil-livefront noticed that removing the axes ternary ended up with the same result? Curious if you tried that.

```
        OffsetObservingScrollView(
            axes: .vertical,
            offset: .init(
                get: { store.state.scrollOffset },
                set: { store.send(.scrollOffsetChanged($0)) }
            )
        ) {
```

File: BitwardenShared/UI/Auth/ProfileSwitcher/ProfileSwitcherView.swift (hunk 1 above)
Comment (fedemkr) [reply BitwardenShared/UI/Auth/ProfileSwitcher/ProfileSwitcherView.swift:64]: It doesn't seem to be necessary, at least on the iOS 18 device.

File: BitwardenShared/UI/Auth/ProfileSwitcher/ProfileSwitcherView.swift (hunk 2 above)
Comment (fedemkr) [reply BitwardenShared/UI/Auth/ProfileSwitcher/ProfileSwitcherView.swift:12]: Yep, that also works. But if we're not changing the `axes` and not going to use the `scrollOffset` then I'd just use a normal `ScrollView` for this as the `OffsetObservingScrollView` won't add any value to this.

Comment (matt-livefront) [reply BitwardenShared/UI/Auth/ProfileSwitcher/ProfileSwitcherView.swift:12]: I guess I'm confused why we don't need the `scrollOffset` anymore?

Comment (matt-livefront) [reply BitwardenShared/UI/Auth/ProfileSwitcher/ProfileSwitcherView.swift:12]: Maybe there's an issue with the scroll offset currently. But I think the intention of it was to hide the space above the scroll view if you scroll it beyond the top of the content.

| Current | Expected |
| -- | -- |
| <video src="https://github.com/user-attachments/assets/087b67fd-7134-4c2f-9b3f-4af8f72b2694" /> | <video src="https://github.com/user-attachments/assets/f041870a-9234-4826-9dcc-aadde6b226a6" /> |

I can fix this separately, but if it's not causing any issue maybe we should keep it in.

**Review by matt-livefront — DISMISSED** [review]

**Comment by fedemkr** [comment]
@matt-livefront had to remake the snapshots for the profile switcher because now it doesn't show the blank space behind the toolbar. I think it's ok and I've just overridden the old snapshots with the current behavior. Let me know if you think this could cause an issue somewhere but I can't think of a place where this could go wrong.

**Comment by matt-livefront** [comment]
> @matt-livefront had to remake the snapshots for the profile switcher because now it doesn't show the blank space behind the toolbar. I think it's ok and I've just overridden the old snapshots with the current behavior. Let me know if you think this could cause an issue somewhere but I can't think of a place where this could go wrong.

Ah, I think that's ok. The gap at the top I think is from the safe area, and switching from the VStack to ZStack likely changed the background in the safe area. But since the profile switcher will always display below a navigation bar, I don't think it's an issue.

**Review by matt-livefront — APPROVED** [review]

**Review by matt-livefront — DISMISSED** [review]

**Review by KatherineInCode — DISMISSED** [review]

**Review by KatherineInCode — APPROVED** [review]

**Pull request merged**